inf-clojure.el

```diff
@@ -688,8 +708,12 @@ If `comint-use-prompt-regexp' is nil (the default), \\[comint-insert-input] on
   (setq comint-input-sender 'inf-clojure--send-string)
   (setq comint-prompt-regexp inf-clojure-comint-prompt-regexp)
   (setq mode-line-process '(":%s"))
-  (clojure-mode-variables)
-  (clojure-font-lock-setup)
+  ;; We don't need to call it in Clojure buffers.
+  (unless (inf-clojure--clojure-buffer-p)
+    ;; Using Tree-sitter based syntax highlighting in comint buffer is
```

Comment:
Ah, this was a research item on my todo list. What's the blocker there currently? (I assumed the parser could be constrained to the current prompt or something along those lines)

Reply:
Well, either we should expose some API from `clojure-ts-mode` or bring some functions from `treesit.el` to `inf-clojure`.

Reply:
I think that'd be perfectly fine, as ideally we should get to the point where there's no hard dep on `clojure-mode` in both CIDER and inf-clojure.

Reply:
Sounds good. But this will require some extra research, I tried to copy some font-lock-* related calls from `treesit.el` and it didn't work out. I'll add a note to handle it separately.

Also it looks like this package should be compatible with at least Emacs 26.2, I believer treesit.el is not available in this version yet.

Reply:
FTR, I made it work by calling the following functions:

```emacs-lisp
(treesit-parser-create 'clojure)
(clojure-ts-mode-variables nil nil)
(treesit-major-mode-setup)
```

![image](https://github.com/user-attachments/assets/19c8d87c-b894-4bf2-891b-b998fd377d96)

The more history in the REPL buffer, the bigger is the syntax tree. Not sure if that might be a problem.

Reply:
Probably it won't be a bit issue in practice. 

P.S. It'd be nice to apply something similar to CIDER's REPL as well.